mailbox/noise.go

```diff
@@ -662,33 +655,23 @@ func (b *Machine) GenActTwo() ([ActTwoSize]byte, error) {
 	//
 	// However, we also want to send the client data they may need for
 	// authentication (if present) encrypted with strong forward secrecy.
-	var payload [ActTwoPayloadSize]byte
-	if b.authData != nil {
-		// If we have an auth payload, then we'll write out 2 bytes
-		// that denotes the true length of the payload, followed by the
-		// payload itself.
-		var payloadWriter bytes.Buffer
-
-		var length [2]byte
-		payLoadlen := len(b.authData)
-		binary.BigEndian.PutUint16(length[:], uint16(payLoadlen))
-
-		if _, err := payloadWriter.Write(length[:]); err != nil {
-			return actTwo, err
-		}
-		if _, err := payloadWriter.Write(b.authData); err != nil {
-			return actTwo, err
-		}
-
-		copy(payload[:], payloadWriter.Bytes())
+	var payloadWriter bytes.Buffer
+	if _, err := payloadWriter.Write(b.authData); err != nil {
+		return nil, err
 	}
+	authPayload := b.EncryptAndHash(payloadWriter.Bytes())
 
-	authPayload := b.EncryptAndHash(payload[:])
+	// We'll write out 2 bytes that denotes the true length of the
+	// authentication payload, followed by the payload itself.
+	var payloadLen [2]byte
+	binary.BigEndian.PutUint16(payloadLen[:], uint16(len(authPayload)))
```

Comment:
Any reason to go with `uint16` and not `uint32` directly? This would limit us to 65k of payload which might be small if we're adding a bunch of permissions and caveats to a macaroon.

Reply:
Also, we should probably use a constant for the `2` here, just so we can make this change more easily without forgetting it in some place.